### DISCUSSION TITLE
WaitForReplicationAfterSaveChanges question

### DISCUSSION THREAD

**Discussion by JezhikLaas** [ravendb/ravendb#13594] [Q&A]
Hello,
what we would like to achieve: replicate to all available (ignoring inaccessable) nodes before returning from SaveChanges, so that following round robin reads always see the latest data. Is this possible? As far as I understand WaitForReplicationAfterSaveChanges it will make SaveChanges fail if a node is down and the given number of replicas cannot be reached, putting the whole cluster into a read only state effectively. Or am I wrong here? Any other ideas?

Thanks, Uwe

**Comment by ayende** [comment]
`WaitForReplicationAfterSaveChanges` will return once the node has replicated to the provided number of nodes.

The failure modes are thus:

* We failed to talk to the original node - nothing was written.
* We were able to write to the original node, but failed to replicate to the required number of siblings in the given time - the data _was_ written and will be replicated in the future.
* The data was written and replicated, but there was an error returning the successful result to the caller. 

In *all* cases, there is no such thing as putting the whole cluster into a read only state. You'll simply have a single request fail. 
Other requests, and any future writes, will work normally.
